cli/src/pcluster/api/util.py

```diff
@@ -177,19 +177,47 @@ def _assert_node_executable():
 def _assert_node_version():
     try:
         # A nosec comment is appended to the following line in order to disable the B607 and B603 checks.
-        # [B607:start_process_with_partial_path] Is suppressed because location of executable is retrieved from env PATH
+        # [B607:start_process_with_partial_path] Is suppressed because location of executable is retrieved from env
+        #   PATH
         # [B603:subprocess_without_shell_equals_true] Is suppressed because input of check_output is not coming from
         #   untrusted source
-        node_version = subprocess.check_output(  # nosec
+        node_version_string = subprocess.check_output(  # nosec
             ["node", "--version"], stderr=subprocess.STDOUT, shell=False, encoding="utf-8"
         )
-        LOGGER.debug("Found Node.js version (%s)", node_version)
+        LOGGER.debug("Found Node.js version (%s)", node_version_string)
     except Exception:
-        message = "Unable to check Node.js version"
+        LOGGER.debug("Unable to determine current Node.js version from node")
+        try:
+            # A nosec comment is appended to the following line in order to disable the B607 and B603 checks.
+            # [B607:start_process_with_partial_path] Is suppressed because location of executable is retrieved from env
+            #   PATH
+            # [B603:subprocess_without_shell_equals_true] Is suppressed because input of check_output is not coming from
+            #   untrusted source
+            node_version_string = subprocess.check_output(  # nosec
+                ["nvm", "current"], stderr=subprocess.STDOUT, shell=False, encoding="utf-8"
+            )
+            LOGGER.debug("Found Node.js version '%s' in use", node_version_string)
+        except Exception:
+            message = "Unable to check Node.js version"
+            LOGGER.critical(message)
+            raise Exception(message)
+        if node_version_string == "none":
```

Comment:
[optional] might be useful to tell in a comment what command we expect to return the "none" string. is it `nvm current`? do we need a test case for this?

Reply:
I think this [test case](https://github.com/aws/aws-parallelcluster/pull/4493/files#diff-9350f46969010c1d42950e607a45080fda852e9c43b1939bd821b40212586734R59) covers the none response? Let me know if I'm missing something though.

Reply:
Added comment.